plugins/com.gwtplugins.gdt.eclipse.suite/src/com/google/gdt/eclipse/suite/wizards/WebAppTemplateProjectCreator.java

```diff
@@ -0,0 +1,435 @@
+
+        // Set the default output directory
+        WebAppUtilities.setOutputLocationToWebInfClasses(createdJavaProjects[i], monitor);
+        /**
+         * Copy files into the web-inf lib folder. This code assumes that it is running in a context that has a workspace
+         * lock.
+         */
+        Sdk gwtSdk = getGWTSdk();
+        setGwtSdk(gwtSdk);
+        if (gwtSdk != null) {
+          new GWTUpdateWebInfFolderCommand(createdJavaProjects[i], gwtSdk).execute();
+        }
+      }
+      reformatJavaFiles(dir);
+
+      // Allow other extensions to run after project creation
+      includeExtensionPartipants();
+    }
+  }
+
+  /**
+   * Replaces all occurrences of the PACKAGENAME with the given package name.
+   * @param iProject
+   * @param dir
+   * @throws IOException
+   */
+  private void replacePackageNames(IProject iProject, File dir) throws IOException {
+    Stack<File> files = new Stack<>();
+    files.add(dir);
+    while(files.isEmpty() == false)
+    {
+      File file = files.pop();
+      if(file.getName().contains("_PACKAGENAME_"))
+      {
+        String name = file.getName();
+        name = name.replace("_PACKAGENAME_", packageBaseName.replace('.', File.separatorChar));
+        File newFile = new File(file.getParentFile(), name);
+        newFile.getParentFile().mkdirs();
+        file.renameTo(newFile);
+        file = newFile;
+      }
+      if(file.getName().contains("_.._"))
+      {
+        String name = file.getName();
+        name = name.replace("_.._", "../");
+        File newFile = new File(file.getParentFile(), name);
+        newFile.getParentFile().mkdirs();
+        file.renameTo(newFile);
+        file = newFile;
+      }
+      if(file.isDirectory())
+      {
+        File[] cf = file.listFiles();
+        files.addAll(Arrays.asList(cf));
+      }
+      else
+      {
+        replaceInFile(file.toString(), "\\$\\{PACKAGENAME\\}", packageBaseName);
+        replaceInFile(file.toString(), "\\$\\{PROJECTNAME\\}", projectBaseName);
+      }
+    }
+  }
+
+  /**
+   * Replaces a pattern in a File. The File is not allowed to be larger then 1.000.000 bytes.
+   * @param file the file, where the replacement is made.
+   * @param pattern a regular expression to be replaced in the whole file.
+   * @param replacement the replacement Text.
+   * @throws IOException
+   */
+  public static void replaceInFile(String file, String pattern, String replacement) throws IOException {
+    Charset cset = Charset.defaultCharset();
+    java.nio.file.Path path = Paths.get(file);
+    if (Files.exists(path)) {
+      if (Files.size(path) > 1000000) {
+        throw new IOException("File to large");
+      }
+      byte[] data = Files.readAllBytes(path);
+      String str = new String(data, cset);
+      String newStr = str.replaceAll(pattern, replacement);
+      BufferedWriter writer = Files.newBufferedWriter(path, cset);
+      writer.write(newStr);
+      writer.close();
```

Comment:
try-with-resources, otherwise if the write fails this might not get closed (and then you've got a broken project, and windows won't let it be deleted or the like)

(goes away if you use Files.write())

Reply:
changed to Files.writeString()